include/unifex/retry_when.hpp

```diff
@@ -355,19 +351,17 @@ class _sender<Source, Func>::type {
         (Source&&)source_, (Func&&)func_, (Receiver&&)r};
   }
 
-  template<
-    typename Receiver,
-    std::enable_if_t<
-        std::is_copy_constructible_v<Source> &&
-        std::is_copy_constructible_v<Func> &&
-        std::is_constructible_v<std::remove_cvref_t<Receiver>, Receiver> &&
-        is_connectable_v<Source&, source_receiver<Source, Func, std::remove_cvref_t<Receiver>>>, int> = 0>
-  operation<Source, Func, Receiver> connect(Receiver&& r) const&
-      noexcept(
-        std::is_nothrow_copy_constructible_v<Source> &&
-        std::is_nothrow_copy_constructible_v<Func> &&
-        std::is_nothrow_constructible_v<std::remove_cvref_t<Receiver>, Receiver> &&
-        is_nothrow_connectable_v<Source&, source_receiver<Source, Func, std::remove_cvref_t<Receiver>>>) {
+  UNIFEX_TEMPLATE(typename Receiver)
+    (requires constructible_from<Source, const Source&> &&
+        constructible_from<Func, const Func&> &&
+        constructible_from<std::remove_cvref_t<Receiver>, Receiver> &&
+        sender_to<
+            Source&,
+            source_receiver<Source, Func, std::remove_cvref_t<Receiver>>>)
+  operation<Source, Func, Receiver> connect(Receiver&& r) const &
+      noexcept(std::is_nothrow_constructible_v<
+          operation<Source, Func, Receiver>, const Source&, const Func&, Receiver> &&
+          is_nothrow_connectable_v<Source&, source_receiver<Source, Func, std::remove_cvref_t<Receiver>>>) {
```

Comment:
I would like a nothrow trait that is directly relatable to sender_to in the requires expression..

Reply:
It's not clear to me that that's an improvement. `sender_to` is a cluster of requirements. If we had an `is_nothrow_sender_to`, which required expression should be noexcept? All of them, it just `connect`? Why?